build.sbt

```diff
@@ -11,12 +11,13 @@ lazy val root = (project in file(".")).
     )),
     name := "scalatest-example",
     Compile / PB.targets := Seq(
-      scalapb.gen() -> (Compile / sourceManaged).value / "scalapb"
+      scalapb.gen(FlatPackage) -> (Compile / sourceManaged).value / "scalapb",
+      PB.gens.java -> (Compile / sourceManaged).value / "java"
```

Comment:
You are using the Java protoc-gen-validate plugin below, in addition to the Scala validate one, so we need to genreate Java sources too.